debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -318,6 +299,89 @@ protected MongoDbOffsetContext initializeOffsets(MongoDbConnectorConfig connecto
                 new MongoDbIncrementalSnapshotContext<>(false), positions);
     }
 
+    /**
+     * A factory to produce a MongoDB change stream pipeline expression.
+     */
+    static class ChangeStreamPipelineFactory {
+
+        private final MongoPreferredNode mongo;
+        private final ReplicaSetOffsetContext rsOffsetContext;
+        private final MongoDbConnectorConfig connectorConfig;
+
+        ChangeStreamPipelineFactory(MongoPreferredNode mongo, ReplicaSetOffsetContext rsOffsetContext, MongoDbConnectorConfig connectorConfig) {
+            this.mongo = mongo;
+            this.rsOffsetContext = rsOffsetContext;
+            this.connectorConfig = connectorConfig;
+        }
+
+        List<Bson> create() {
+            // Resolve the leaf filters
+            var filters = Stream
+                    .of(
+                            createCollectionFilter(mongo),
+                            createOperationTypeFilter(connectorConfig),
+                            createClusterTimeFilter(rsOffsetContext))
+                    .flatMap(Optional::stream)
+                    .collect(toList());
+
+            // Combine
+            var andFilter = Filters.and(filters);
+            var matchFilter = Aggregates.match(andFilter);
+            var pipeline = Collections.singletonList(matchFilter);
+            LOGGER.debug("Change stream pipeline: {}", pipeline);
+
+            return pipeline;
+        }
+
+        private static Optional<Bson> createCollectionFilter(MongoPreferredNode mongo) {
+            // Get collection filters on change stream pipeline. Note that `MongoPreferredNode#collections`
+            // performs the configured filtering by regex
+            // See https://www.mongodb.com/docs/manual/reference/change-events/create/#mongodb-data-create
+            var nsList = mongo.collections().stream()
+                    .map(c -> new Document()
+                            .append("db", c.dbName())
+                            .append("coll", c.name()))
+                    .collect(toList());
+
+            return Optional.of(Filters.in("ns", nsList));
+        }
+
+        private static Optional<Bson> createOperationTypeFilter(MongoDbConnectorConfig connectorConfig) {
+            var skippedOperations = connectorConfig.getSkippedOperations();
+            if (skippedOperations.isEmpty()) {
+                // Include all
+                return Optional.empty();
+            }
+
+            // TODO: What about the rest of these? https://www.mongodb.com/docs/manual/reference/change-events/
```

Comment:
I was scanning through the above link and noticed that not all events are represented. I'm wondering if it makes more sense to convert this expression a `$nin` filter instead?